### PR TITLE
fix: story mention

### LINKED ISSUE #5258: Messenger Integration: Content from Instagram stores permanently
**Describe the bug**

When a user mention the connected Instagram page in a story, the content of the story is downloaded in Chatwoot, then if the user deletes the story content persists in the platform. 
**To Reproduce**

1. Go to Instagram with the tester account, add a story and mention the page that is connected to Chatwoot
2. You will see the notification message in the Messenger channel
3. Delete the story in Instagram
4. You will see that the content from the story is still stored in Chatwoot

**Expected behavior**

The temporary content from Instagram shouldn't be stored in the platform. This is neccesary to get the aprobal from Facebook.

**Screenshots**

image
image

**Environment**

Self hosted installation running in a Linux virtual machine.

**Additional context**

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #5258 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
This is blocked as we are not able to test on staging.

**Comment by tejaswinichile** [comment]
Uploading Screen Recording 2022-11-23 at 2.16.55 PM.mov…

@sojan-official This is working well with the live mode.
 
I will be updating the relative documentation over the weekend or may be next week

**Review by sojan-official — COMMENTED** [review]
- a couple of comments

File: app/models/message.rb [hunk 1]
```diff
@@ -106,10 +106,27 @@ def push_event_data
...
+
+    if result['story']['mention']['link'].blank?
+      attachments.destroy_all
+      update(
+        content: I18n.t('conversations.messages.instagram_deleted_story_content'),
+        content_attributes: {}
+      )
+      reload
+    end
+  end
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply app/models/message.rb:128]: This happens when we fetch the `push_event_data` for the message, and we need to update the content of the message according to story-mention is present or not.

File: app/models/attachment.rb [hunk 2]
```diff
@@ -129,4 +133,41 @@ def validate_file_size(byte_size)
...
+
+  def delete_instagram_story(message)
+    message.update(content: I18n.t('conversations.messages.instagram_deleted_story_content'))
+    delete
+  end
+
+  def add_ig_story_data_url(metadata)
+    metadata[:data_url] = external_url
+    metadata[:thumb_url] = external_url
+    metadata
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply app/models/attachment.rb:171]: I think fetch_story_link should be move to the `facebook::page` and update and destroy we will keep it here.

File: app/models/channel/facebook_page.rb [hunk 3]
```diff
@@ -63,4 +63,20 @@ def unsubscribe
     Rails.logger.debug { "Rescued: #{e.inspect}" }
     true
   end
+
+  # TODO: We will be removing this code after instagram_manage_insights is implemented
+  def fetch_instagram_story_link(message)
+    k = Koala::Facebook::API.new(page_access_token)
+    result = k.get_object(message.source_id, fields: %w[story]) || {}
+    story_link = result['story']['mention']['link']
+    delete_instagram_story(message) if story_link.blank?
```
[the remark this answers is not in the document]
Comment (tejaswinichile) [reply app/models/channel/facebook_page.rb:72]: If the story is expired then it raises the ClientError and if the story is deleted with valid story-id it responses with `nil`

**Review by sojan-official — COMMENTED** [review]
- [x] factory for story mention
- [ ] spec for  attachment push event data

File: spec/factories/channel/instagram_channel.rb [hunk 4]
```diff
@@ -6,5 +6,9 @@
     user_access_token { SecureRandom.uuid }
     page_id { SecureRandom.uuid }
     account
+
+    before :create do |_channel|
+      WebMock::API.stub_request(:post, 'https://graph.facebook.com/v3.2/me/subscribed_apps')
+    end
```
Comment (tejaswinichile) [inline spec/factories/channel/instagram_channel.rb:12]: 💯

**Pull request merged**